src/main/java/org/nypl/findawayclientlibrary/DownloadService.java

```diff
@@ -91,11 +108,33 @@ public void initDownloadEngine() {
 
   /**
    * Subscribe to a stream of _all_ download events for the supplied content id.
+   * Keep the record of the subscription, in case will be asked to unsubscribe.
+   *
+   * NOTE: Download events are described here:  http://developer.audioengine.io/sdk/android/v7/download-engine .
    * @return
    */
   public Subscription subscribeDownloadEventsAll(Observer<DownloadEvent> observer, String contentId) {
-    Subscription eventsSubscription = this.getDownloadEngine().events(contentId).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(observer);
-    return eventsSubscription;
+    // reset
+    if (eventsSubscriptionAll != null && !eventsSubscriptionAll.isUnsubscribed()) {
+      eventsSubscriptionAll.unsubscribe();
+    }
+
+    eventsSubscriptionAll = this.getDownloadEngine().events(contentId).subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread()).subscribe(observer);
```

Comment:
It seems like you can easily have a null pointer exception here if `this.getDownloadEngine()` returns `null`.

Reply:
fixed